pennylane/devices/default_qubit.py

```diff
@@ -66,6 +66,14 @@ def stopping_condition(op: qml.operation.Operator) -> bool:
         return True
     if op.__class__.__name__[:3] == "Pow" and qml.operation.is_trainable(op):
         return False
+    if (
+        isinstance(op, qml.operation.Operator)
+        and not isinstance(op, SymbolicOp)
+        and op.has_sparse_matrix
+        and (not op.has_matrix)
+        and len(op.wires) >= 3
+    ):
+        return True
```

Comment:
How does the stop condition relate to data representation?
There is no decomposition for matrix-only operators, right?

Reply:
Here the idea is we want to filter those who have strictly sparse representation (e.g. QubitUnitary) s.t. they don't get caught by preprocessor to try to find the decomposition (they don't have).
However certain symbolic operators e.g. AQFT in https://github.com/PennyLaneAI/pennylane/actions/runs/13574496901/job/37947325351 have the same property pattern, therefore we ignored them in this addtional stopping condition.

Reply:
But why `and len(op.wires) >= 3`?

Reply:
one-site or two-site operators can be decomposed just as usual